indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamSamplerSpec.java

```diff
@@ -92,9 +93,22 @@ public SamplerResponse sample()
       );
       inputFormat = null;
     } else {
+      RecordSupplier<PartitionIdType, SequenceOffsetType, RecordType> recordSupplier;
+
+      try {
+        recordSupplier = createRecordSupplier();
+      }
+      catch (Exception e) {
```

Comment:
I think we should log the stack trace here instead of doing it in `toResponse`. This way, it is being logged closer to where the exception is getting generated from.

Reply:
There're also other code that throw SamplerException, so I think `SamplerExceptionMapper` is a good place to handle the exception.